### PR TITLE
Feature/arnav jacob/page footer

### LINKED ISSUE #4: Page Footer
Description: Create a footer that contains links to each page within the web app, as well as a newsletter subscription box. Create a textfield + button component with input validation that will check if the submitted email address follows an email domain. If so, make a POST request to the backend to send the email address. If not, display a message to inform the user to enter a valid email address.

Task Type: Frontend, Backend

image

### PR DESCRIPTION
**Pull request by jackavh**
## Tracking Info

Resolves #4 

### Frontend

- Created the footer and added styling based on figma to files footer.tsx and Footer.module.css
- Linked to the backend API to provide feedback on newsletter submission: invalid email / duplicate email

### Backend

- Created a schema for subscribers in the mongoDB table
- Wrote validation chain that normalizes and validates submitted email addresses
- Performs check to identify duplicate email submissions

## Testing

- Checked that the design of the frontend matches the backend
- Checked that MongoDB was getting updated with email addresses (for the newsletter)
- Ran a series of test emails invalid and duplicate to make sure that validation works

## Confirmation of Change

Screenshot 2024-01-30 at 4 39 21 PM

Screenshot 2024-01-30 at 4 39 36 PM

Screenshot 2024-01-30 at 4 39 52 PM

Screenshot 2024-01-30 at 4 40 08 PM

### CODE REVIEW

File: frontend/src/components/footer.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+          </form>
+        </div>
+
+        {/* LINK QUAD */}
+        <div className={styles.linkDiv}>
+          <div className={styles.linkRow}>
+            <Link className={`${styles.link} ${styles.boldLink}`} href={links.aboutUs}>
+              About Us
+            </Link>
+            <a className={`${styles.link} ${styles.boldLink}`} href={links.getInvolved}>
```
Comment (jennymar) [inline frontend/src/components/footer.tsx]: We probably want to be using Link tags for all link between pages in the web app

**Review by jennymar — APPROVED** [review]
LGTM! I pushed some code for adding add the Footer component to layout.tsx which will apply automatically the footer to every page inside the app directory, as well as moved some fonts over to globals.css so they can be reused. Great work!

File: frontend/src/components/footer.tsx (hunk 1 above)
Comment (jackavh) [reply frontend/src/components/footer.tsx]: Pushed a fix for this, see Slack

**Review by kevindo0720 — COMMENTED** [review]
Looks good to me

**Review by kevindo0720 — APPROVED** [review]

**Pull request merged**